Random Users/View Controllers/UsersTableViewController.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  UsersTableViewController.swift
+//  Random Users
+//
+//  Created by ronald huston jr on 9/12/20.
+//  Copyright © 2020 Erica Sadun. All rights reserved.
+//
+
+import UIKit
+
+class UsersTableViewController: UITableViewController {
+
+    //  MARK: - properties
+    let userClient = UserClient()
+    
+    let cache = Cache<String, Data>()
+    private let queue = OperationQueue()
+    private var operations = [String : Operation]()
+    
+    var users: [User]? {
+        didSet {
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        userClient.fetchUsers { (users, error) in
+            if let error = error {
+                print("error performing data task: \(error)")
+                return
+            }
+            self.users = users
+        }
+    }
+
+    // MARK: - Table view data source
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        
+        return users?.count ?? 0
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "UserCell", for: indexPath) as! UserTableViewCell
+        guard let user = users?[indexPath.row] else { return cell }
+        cell.nameLabel.text = user.name
+        loadImage(forCell: cell, forItemAt: indexPath)
+
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, didEndDisplaying cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        guard let user = users?[indexPath.row],
+            let email = user.email else { return }
+        
+        operations[email]?.cancel()
+    }
+
+    
+    // MARK: - Navigation
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "UserDetail" {
+            
+        }
+    }
+
+
+    private func loadImage(forCell cell: UserTableViewCell, forItemAt indexPath: IndexPath) {
+        guard let user = users?[indexPath.row],
+            let email = user.email else { return }
+        
+        if let data = cache.value(key: email) {
+            cell.userImageView?.image = UIImage(data: data)
+            
+        } else {
+            let fetchOperation = FetchImageOperation(userRandom: user)
+            let cacheOperation = BlockOperation {
+                guard let data = fetchOperation.imageData else { return }
+                self.cache.cache(key: email, value: data)
+            }
+            
+            let displayImageOperation = BlockOperation {
+                defer { self.operations.removeValue(forKey: email)}
+                
+                if let data = fetchOperation.imageData {
+                    cell.imageView?.image = UIImage(data: data)
```

Comment:
This actually came up with a bunch of students this sprint, but here you are setting the cells imageView property to the users image, not entirely sure why this works but it does, but this is why your images don't show up until you start scrolling (not entirely sure why you need to scroll for them to show up), but change it to match the name of the imageView you created on your custom cell, like this...
```suggestion
                    cell.userImageView.image = UIImage(data: data)
```
You actually used the correct image view in the above code where its checking if its stored in the cache.